flair/embeddings.py

```diff
@@ -326,6 +326,84 @@ def extra_repr(self):
+
+    def __init__(self, model_file: str):
+        super().__init__()
+
+        self.name = 'elmo-transformer'
+        self.static_embeddings = True
+        self.allen_nlp_utils = ELMoTransformerEmbeddings.AllenNlpUtils(model_file)
+
+        # embed a dummy sentence to determine embedding_length
+        dummy_sentence: Sentence = Sentence()
+        dummy_sentence.add_token(Token('hello'))
+        embedded_dummy = self.embed(dummy_sentence)
+        self.__embedding_length: int = len(embedded_dummy[0].get_token(1).get_embedding())
+
+    @property
+    def embedding_length(self) -> int:
+        return self.__embedding_length
+
+    class AllenNlpUtils:
+        def __init__(self, model_file):
+            from allennlp.modules.token_embedders.bidirectional_language_model_token_embedder import \
+                BidirectionalLanguageModelTokenEmbedder
+            from allennlp.data.token_indexers.elmo_indexer import ELMoTokenCharactersIndexer
+
+            self.lm_embedder = BidirectionalLanguageModelTokenEmbedder(
+                archive_file=model_file,
+                dropout=0.2,
+                bos_eos_tokens=["<S>", "</S>"],
+                remove_bos_eos=True,
+                requires_grad=False
+            )
+
+            if torch.cuda.is_available():
+                device = torch.device('cuda')
+            else:
+                device = torch.device('cpu')
+
+            self.device = device
+            self.lm_embedder = self.lm_embedder.to(device=device)
+            self.vocab = self.lm_embedder._lm.vocab
+            self.indexer = ELMoTokenCharactersIndexer()
+
+        def __getstate__(self):
+            # We need to overwrite the state dictionary to avoid the following error:
+            # AttributeError: Can't pickle local object 'TokenCharactersEncoder.__init__.<locals>.<lambda>'
+            pass
+
+    def _add_embeddings_internal(self, sentences: List[Sentence]) -> List[Sentence]:
+        # Avoid conflicts with flair's Token class
+        import allennlp.data.tokenizers.token as allen_nlp_token
+
+        indexer = self.allen_nlp_utils.indexer
+        vocab = self.allen_nlp_utils.vocab
+
+        for sentence in sentences:
+            character_indices = indexer.tokens_to_indices([allen_nlp_token.Token(token.text) for token in sentence],
+                                                              vocab, "elmo")["elmo"]
+
+            indices_tensor = torch.LongTensor([character_indices])
+            indices_tensor = indices_tensor.to(device=self.allen_nlp_utils.device)
+            embeddings = self.allen_nlp_utils.lm_embedder(indices_tensor)[0].detach().cpu().numpy()
+
+            for token, token_idx in zip(sentence.tokens, range(len(sentence.tokens))):
+                token: Token = token
+                embedding = embeddings[token_idx]
+                word_embedding = torch.autograd.Variable(torch.FloatTensor(embedding))
```

Comment:
`Tensor` and `Variable` was merged into one class with PyTorch 0.4 (see https://pytorch.org/blog/pytorch-0_4_0-migration-guide/#merging-tensor-and-variable-and-classes). 
So can't you just do
```
word_embedding = torch.FloatTensor(embedding)
```

Reply:
Thanks :) I fixed that (and it works without problems both on CPU and GPU)!